SwanCustomEnvironments/swancustomenvironments/scripts/makenv.sh

```diff
@@ -0,0 +1,164 @@
+    fi
+}
+
+# Function for printing the help page
+print_help() {
+    _log "Usage: makenv --env/-e NAME --repo/-r REPOSITORY [--accpy ACCPY_VERSION] [--help/-h]"
+    _log "Options:"
+    _log "  -e, --env NAME              Name of the custom virtual environment (mandatory)"
+    _log "  -r, --repo REPOSITORY       Path or http link for a public repository (mandatory)"
+    _log "  -h, --help                  Print this help page"
+    _log "  --accpy VERSION             Version of Acc-Py to be used"
+}
+
+# --------------------------------------------------------------------------------------------
+
+# Parse command line arguments
+while [ $# -gt 0 ]; do
+    key="$1"
+    case $key in
+        --env|-e)
+            ENV_NAME=$2
+            shift
+            shift
+            ;;
+        --repo|-r)
+            REPOSITORY=$2
+            shift
+            shift
+            ;;
+        --accpy)
+            ACCPY_VERSION=$2
+            shift
+            shift
+            ;;
+        --help|-h)
+            print_help
+            exit 0
+            ;;
+        *)
+            >&2 _log "ERROR: Invalid argument: $1" && _log
+            print_help
+            exit 1
+            ;;
+    esac
+done
+
+# --------------------------------------------------------------------------------------------
+
+# Checks if a name for the environment is given
+if [ -z "$ENV_NAME" ]; then
+    >&2 _log "ERROR: No virtual environment name provided." && _log
+    print_help
+    exit 1
+fi
+
+# Checks if the provided Acc-Py version is valid
+if [ -n "$ACCPY_VERSION" ] && [ ! -e "$ACCPY_PATH/base/$ACCPY_VERSION" ]; then
+    >&2 _log "ERROR: Invalid Acc-Py version."
+    exit 1
+fi
+
+# Checks if a repository file is given
+if [ -z "$REPOSITORY" ]; then
+    >&2 _log "ERROR: No repository provided." && _log
+    print_help
+    exit 1
+# Checks if the provided repository source is found
+elif [ -d $REPOSITORY ]; then
+    REQ_PATH="${REPOSITORY}/requirements.txt"
+    if [ ! -f "${REQ_PATH}" ]; then
+        >&2 _log "ERROR: Requirements file not found (${REQ_PATH})."
+        exit 1
+    fi
+elif [[ $REPOSITORY == http* ]]; then
+    # Extract the repository name from the URL
+    repo_name=$(basename $REPOSITORY)
+    repo_name=${repo_name%.*}
+    
+    mkdir -p $HOME/SWAN_projects
+
+    define_repo_path $repo_name
+
+    # Clone the repository
+    echo "Cloning the repository from ${REPOSITORY}..."
+    git clone $REPOSITORY -q "${REPO_PATH}" || { >&2 _log "ERROR: Failed to clone repository"; exit 1; }
+
+    REQ_PATH=${REPO_PATH}/requirements.txt
+    # Check if requirements.txt exists in the repository
+    if [ ! -f "${REQ_PATH}" ]; then
+        rm -rf ${REPO_PATH}
+        >&2 _log "ERROR: ${REQ_PATH} not found in ${REPO_PATH}."
+        exit 1
+    fi
+else
+    >&2 _log "ERROR: Invalid repository (${REPOSITORY})."
+    exit 1
+fi
+
+
+# --------------------------------------------------------------------------------------------
```

Comment:
Same comment here, if you are doing this separation of the code in the script based on some criteria, it's better to be explicit.